### DISCUSSION TITLE
etcd backend certificates

### DISCUSSION THREAD

**Discussion by omeyer99** [minio/minio#13838] [Q&A]
Hey, I'm trying to configure my minIO-Instance running on kubernetes with an etcd backend. since my etcd-cluster uses tls authentication with self-signed certificates, i cannot connect my minIO-Instance to my etcd-cluster. normally i would have to submit the ca certificates as well, but that is not possible in the current minio version or am i missing something. 
thanks for the help.

`curl -v --key srv/minio/etcd/client-key.pem --cert srv/minio/etcd/client.pem https://etcd.example.com:2379` - does not work
`curl -v --cacert srv/minio/etcd/ca.pem --key srv/minio/etcd/client-key.pem --cert srv/minio/etcd/client.pem https://etcd.example.com:2379` - works

**Comment by harshavardhana** [comment]
> `curl -v --key srv/minio/etcd/client-key.pem --cert srv/minio/etcd/client.pem https://etcd.example.com:2379` - does not work
> `curl -v --cacert srv/minio/etcd/ca.pem --key srv/minio/etcd/client-key.pem --cert srv/minio/etcd/client.pem https://etcd.example.com:2379` - works

how is this MinIO's problem? @omeyer99

**Reply by omeyer99** [reply]
It's a minio problem because I want to run my minio instance with an etc backend, but this doesn't work because I can't pass the CA certificate to the minio instance and thus the connection doesn't come up.
The curl commands were just to illustrate the problem.

**Reply by omeyer99** [reply]
Let me specify it in more detail. 
I followed the instructions and set the configuration options using Environment Variables:
`MINIO_ETCD_ENDPOINTS, 
MINIO_ETCD_CLIENT_CERT, 
MINIO_ETCD_CLIENT_CERT_KEY`.
Even though the connection parameters are correct, the MinIO instance does not connect to the etcd. My guess was that it is because of the selfsigned certificates. to solve this problem, i had assumed that you have to pass the ca certificate as well

*tested with etcdctl

**Reply by harshavardhana** [reply]
> I followed the instructions and set the configuration options using Environment Variables:
> `MINIO_ETCD_ENDPOINTS, MINIO_ETCD_CLIENT_CERT, MINIO_ETCD_CLIENT_CERT_KEY`.
> Even though the connection parameters are correct, the MinIO instance does not connect to the etcd. My guess was that it is because of the selfsigned certificates. to solve this problem, i had assumed that you have to pass the ca certificate as well

and then you need to put your ca.crt at `~/.minio/certs/CAs`

**Reply by omeyer99** [reply]
This is a very awkward implementation in my eyes, but thanks for the suggested solution.

**Comment by mshanmu** [comment]
@omeyer99 Check this out https://github.com/minio/minio/tree/master/helm/minio#installing-certificates-from-third-party-cas . This link tells you how to install CAs for use by minio server

**Reply by omeyer99** [reply]
Since I wrote my Helm Chart myself, I can't use this solution. Instead I store the CA-certificate under `~/.minio/certs/CAs/`.
